coverage/docs/design.md

```diff
@@ -0,0 +1,139 @@
+#Overview
+This code coverage tool has two major features.
+1. In post-submit workflow (when a pull request is merged), it runs code coverage and generates the 
+  following artifacts in a GCS bucket
+  - xml file that stores file-level and package-level code coverage, formatted to be readable by TestGrid
+  - code coverage profile, which will be used later in presubmit workflow as a base for comparison
+2. In pre-submit workflow (when a pull requested is created or updated with new commit)
```

Comment:
Just mention optional GCS fetch for historical data

Reply:
It's not optional. Without data from postsubmit, the pre-submit workflow here is meaningless. We need the data stored as the basis for comparison